GiniVision/Classes/Networking/AccountingDocumentService.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  AccountingDocumentService.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 1/14/19.
+//
+
+import Foundation
+import Gini_iOS_SDK
+
+final class AccountingDocumentService: DocumentServiceProtocol {
+    var giniSDK: GiniSDK
+    var metadata: GINIDocumentMetadata?
+    var document: GINIDocument?
+    var analysisCancellationToken: BFCancellationTokenSource?
+    
+    init(sdk: GiniSDK, metadata: GINIDocumentMetadata?) {
+        self.giniSDK = sdk
+        self.metadata = metadata
+        self.giniSDK.sessionManager.logIn()
+    }
+    
+    func cancelAnalysis() {
+        if let document = document {
+            delete(document)
+        }
+        
+        analysisCancellationToken?.cancel()
+        resetToInitialState()
```

Comment:
Shouldn't these two lines be inside the if? Just for clarity?

Reply:
No, the analysis can be cancelled when the document is being upload.